adapter/protobuf/src/Mu/Adapter/ProtoBuf.hs

```diff
@@ -353,10 +357,35 @@ instance {-# OVERLAPS #-}
              | otherwise
              = base
 
-instance TypeError ('Text "maps are not currently supported")
+-- see https://developers.google.com/protocol-buffers/docs/proto3#maps
+{-
+message MapFieldEntry {
+  key_type key = 1;
+  value_type value = 2;
+}
+
+repeated MapFieldEntry map_field = N;
+-}
+instance ( KnownNat (FindProtoBufId sch ty name)
+         , ProtoBridgeOneFieldValue sch k
+         , ProtoBridgeOneFieldValue sch v
+         , Ord (FieldValue sch k) )
          => ProtoBridgeField sch ty ('FieldDef name ('TMap k v)) where
-  fieldToProto = error "maps are not currently supported"
-  protoToField = error "maps are not currently supported"
+  fieldToProto (Field (FMap mp))
+    = foldMap oneMapValueToProto (M.toAscList mp)
+    where fieldId = fromInteger $ natVal (Proxy @(FindProtoBufId sch ty name))
+          oneMapValueToProto (k, v)
+            = PBEnc.embedded fieldId $
+                oneFieldValueToProto 1 k <> oneFieldValueToProto 2 v
+  protoToField = Field . FMap . M.fromList <$> go
+    where fieldId = fromInteger $ natVal (Proxy @(FindProtoBufId sch ty name))
+          go = PBDec.repeated (
+                 PBDec.embedded'
+                   ((,) <$> PBDec.one protoToOneFieldValue
+                              (error "key should always be present") `at` 1
+                        <*> PBDec.one protoToOneFieldValue
+                              (error "value should always be present") `at` 2))
```

Comment:
It seems dangerous to throw error here. Maybe this can be handled by using `empty` like [here](https://github.com/higherkindness/mu-haskell/blob/aff0b446e7fce337e6158a918f4b24a489572972/adapter/protobuf/src/Mu/Adapter/ProtoBuf.hs#L317-L318)? 

If that is not possible for some reason, I think a parser error would be better. But it might be wrong, as according to the [docs](https://developers.google.com/protocol-buffers/docs/proto3#backwards_compatibility), this should be compatible with parsing the `MapFieldEntry` message type with key and value, and for that an old encoder may skip encoding an "empty" value in the map.

Reply:
You are completely right. I've fixed this in 67cc414, so now it behaves as any other field: if not present, it tries to use the default value, and otherwise the parsing itself fails.